tasks/main.yml

```diff
@@ -0,0 +1,162 @@
+---
+- name: Install role requirements
+  import_tasks: requirements.yml
+  when: install_home_assistant_supervised_ansible_requirements|bool
+
+- name: Install OS package prereqs
+  apt:
+    name:
+      - systemd
+      - network-manager
+      - apparmor
+      - jq
+      - curl
+      - dbus
+    state: latest
+    update_cache: yes 
+    
+- name: Disable Modem Manager to prevent issues when using serial devices
+  service:
+    name: ModemManager
+    enabled: no
+    state: stopped
+
+- name: Create Docker config directory 
+  file:
+    path: /etc/docker 
+    state: directory
+
+- name: Set up docker config file
+  copy:
+    src: docker_daemon.json
+    dest: /etc/docker/daemon.json
+  notify:
+    - restart docker
+
+- name: Fix kernel dmesg restriction 
+  sysctl:
+    name: kernel.dmesg_restrict 
+    value: '0'
+    sysctl_set: yes
+    state: present
+    reload: yes
+
+- name: Create config for NetworkManager   
+  copy:
+    src: "{{ networkmanager_file['source'] }}" 
+    dest: "{{ networkmanager_file['target'] }}" 
+  loop:
+    - source: NetworkManager.conf
+      target: /etc/NetworkManager/NetworkManager.conf   
+    - source: system-connection-default 
+      target: /etc/NetworkManager/system-connections/default
+  loop_control:
+    loop_var: networkmanager_file
+    label: "{{ networkmanager_file['source'] }}"
+  notify:
```

Comment:
If you are restarting after the next task, is this required? Or can you just restart after the next one and only do it once?

Reply:
Handlers don't run until the end of the role or you hit a `flush_handlers` task. Even if it gets triggered multiple times, the handler only will run once.